vispy/visuals/circle.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright (c) Vispy Development Team. All Rights Reserved.
+# Distributed under the (new) BSD License. See LICENSE.txt for more info.
+# -----------------------------------------------------------------------------
+
+import numpy as np
+import math
+
+from .mesh import MeshVisual
+
+
+class CircleVisual(MeshVisual):
+    """Visual that calculates and displays a circle using triangle's
+
+    Parameters
+    ----------
+    corners : int
+        Data to histogram. Currently only 1D data is supported.
+    radius : int
+        Radius of the circle
+    color : ColorArray
+        ColorArray or color letter for example 'w' for white
+    """
+
+    def __init__(self, corners, radius=1, color='w'):
+
+        color_array = None
+        if isinstance(color, (np.ndarray, np.generic)):
+            if color.shape[0] > 1:
+                color_array = []
+                for i in range(len(color)):
+                    color_array.append(color[i])
+                    if i != (len(color) - 1):
+                        color_array.append([0, 0, 0])
+                color = None
+
+        rr, tris = self._calculate(corners, radius)
+
+        MeshVisual.__init__(self, rr, tris, color=color, face_colors=color_array)
+
+    def update_data(self, corners, radius=1, color='w'):
+
+        color_array = None
+        if isinstance(color, (np.ndarray, np.generic)):
+            if color.shape[0] > 1:
+                color_array = []
+                for i in range(len(color)):
+                    color_array.append(color[i])
+                    if i != (len(color) - 1):
+                        color_array.append([0, 0, 0])
+                color = None
+
+        rr, tris = self._calculate(corners, radius)
+
+        MeshVisual.set_data(self, rr, tris, color=color, face_colors=color_array)
+
+    def _calculate(self, corners, radius):
+        last_angle = 0
+        vertices = []
+        for i in range(corners + 1):
+            if i != 0:
+                y = radius * math.cos(last_angle)
+                x = radius * math.sin(last_angle)
+                vertices.append((x, y))
+                vertices.append((0, 0))
+                angle = 2 * math.pi * i / corners
+                y = radius * math.cos(angle)
+                x = radius * math.sin(angle)
+                vertices.append((x, y))
+                last_angle = angle
```

Comment:
Is there a way to do this without the for loop? For a start, you could remove the `if` and do `range(1, corners + 1)` and that would remove an entire indentation level. Next, if `i` (better variable name?) was `np.arange(corners)` could we do the same type of math and then have `vertices = np.zeros((corners * 3, 3))` and do some level of `vertices[::3, :2] = (radius * np.sin(angles), radius * np.cos(angles))` and then something similar for `vertices[2::3, :2]`. I might have an off-by-one on the size of `i` and need to calculate the `angles`, but otherwise I think that makes sense. Thoughts?

Reply:
I didnt explore it i just needed it quick, and because i'm not very good generationg the verticies arrays i took the easy route. However your suggestion should be a good start,